Translated docs/tutorial/part-six (#13)
gatsbyjs/gatsby-de: docs/tutorial/part-six/index.md

---
title: Transformer-Plugins
typora-copy-images-to: ./
disableTableOfContents: true
---

> > Dieses Tutorial ist Teil einer Reihe über die Datenschicht von Gatsby. Stellen Sie sicher, dass Sie [Teil 4](/tutorial/part-four/) und [Teil 5](/tutorial/part-five/) durchgegangen sind, bevor Sie hier fortfahren.

Reviewer: ```suggestion
> Dieses Tutorial ist Teil einer Reihe über die Datenschicht von Gatsby. Stell bitte sicher, dass du zunächst [Teil 4](/tutorial/part-four/) und [Teil 5](/tutorial/part-five/) durchgegangen bist, bevor du hier fortfährst.
```

## Um was geht es in diesem Tutorial?

> Das vorherige Tutorial zeigte, wie Quellen-Plugins Daten _in_ Gatsbys Datensystem bringen. In diesem Tutorial erfahren Sie, wie Transformer-Plugins den Inhalt von Quellen-Plugins _transformieren_. Die Kombination aus Quellen-Plugins und Transformer-Plugins kann alle Datenquellen und -transformationen verarbeiten, die beim Erstellen einer Gatsby-Site benötigt werden.

Reviewer: ```suggestion
Das vorherige Tutorial hat gezeigt, wie Quellen-Plugins Daten _in_ Gatsbys Datensystem bringen. In diesem Tutorial erfährst du, wie Transformer-Plugins den Inhalt von Quellen-Plugins _transformieren_. Die Kombination aus Quellen-Plugins und Transformer-Plugins kann alle Datenquellen und -transformationen verarbeiten, die beim Erstellen einer Gatsby-Site benötigt werden.
```

## Transformer plugins

> Oft ist das Format der Daten, die Sie von Quellen-Plugins erhalten, nicht das, was Sie brauchen, um Ihre Website zu erstellen. Mit dem Quellen-Plugin des Dateisystems können Sie Daten _über_ Dateien abfragen. Aber was ist, wenn Sie Daten _aus_ Dateien abfragen möchten?

Reviewer: ```suggestion
Oft ist das Format der Daten, die von Quellen-Plugins zurückgegeben werden, nicht das, was du brauchst, um deine Website zu erstellen. Mit dem Dateisystems Quellen-Plugin kannst du Daten _über_ Dateien abfragen. Aber was ist, wenn du Daten _aus_ Dateien abfragen möchtest?
```

> Um dies zu ermöglichen, unterstützt Gatsby Transformer-Plugins, die Raw-Dateien akzeptieren Inhalte aus Quell-Plugins und _transformieren_ sie in etwas Besseres.

Reviewer: ```suggestion
Um dies zu ermöglichen, unterstützt Gatsby Transformer-Plugins welche die rohen Daten von Quellen-Plugins entgegen nehmen und in etwas anderes _transformieren_ können.
```

> Um dies zu ermöglichen, unterstützt Gatsby Transformer-Plugins welche die rohen Daten von Quellen-Plugins entgegen nehmen und in etwas anderes _transformieren_ können.

Reviewer: ```suggestion
Um dies zu ermöglichen, unterstützt Gatsby Transformer-Plugins, welche die Daten von Quellen-Plugins entgegen nehmen und in etwas anderes _transformieren_ können.
```

> Zum Beispiel Abschriften. Markdown ist schön einzuschreiben, aber wenn man einen baut Seite mit, müssen Sie den Markdown HTML sein.

Reviewer: ```suggestion
Zum Beispiel Markdown. Markdown schreibt sich leicht, aber wenn man einen baut Seite, müssen Sie die Markdown Dateien in HTML umgewandelt werden.
```

> Zum Beispiel Markdown. Markdown schreibt sich leicht, aber wenn man einen baut Seite, müssen Sie die Markdown Dateien in HTML umgewandelt werden.

Reviewer: ```suggestion
Zum Beispiel Markdown. Markdown schreibt sich leicht, aber wenn du eine Website baust, müssen Sie die Markdown-Dateien in HTML umgewandelt werden.
```

> Fügen Sie Ihrer Site unter eine Abschriften-Datei hinzu

Reviewer: ```suggestion
Fügen Sie Ihrer Site unter eine Markdown-Datei unter
```

> Fügen Sie Ihrer Site unter eine Markdown-Datei unter

Reviewer: ```suggestion
Füge deiner Website unter eine Markdown-Datei unter
```

> Füge deiner Website unter eine Markdown-Datei unter

Reviewer: ```suggestion
Füge deiner Website eine Markdown-Datei unter
```

> `src/pages/sweet-pandas-eating-sweets.md` (Dies wird zu Ihrem ersten Abschlag Blogeintrag)

Reviewer: ```suggestion
`src/pages/sweet-pandas-eating-sweets.md` hinzu. (Dies wird Ihr erster Markdown Blogeintrag)
```

> `src/pages/sweet-pandas-eating-sweets.md` hinzu. (Dies wird Ihr erster Markdown Blogeintrag)

Reviewer: ```suggestion
`src/pages/sweet-pandas-eating-sweets.md` hinzu. (Dies wird dein erster Markdown Blogbeitrag)
```

> `src/pages/sweet-pandas-eating-sweets.md` hinzu. (Dies wird dein erster Markdown Blogbeitrag)

Reviewer: ```suggestion
`src/pages/sweet-pandas-eating-sweets.md` hinzu (das wird dein erster Markdown Blogbeitrag)
```

> und lernen, wie man es mit Hilfe von Transformer-Plugins und in HTML umwandelt GraphQL.

Reviewer: ```suggestion
und lernen, wie, mit Hilfe von Transformer-Plugins und GraphQL, die Daten in HTML umwandelt werden.
```

> und lernen, wie, mit Hilfe von Transformer-Plugins und GraphQL, die Daten in HTML umwandelt werden.

Reviewer: ```suggestion
und lerne, wie mit Hilfe von Transformer-Plugins und GraphQL, die Daten in HTML umwandelt werden.
```

<iframe width="560" height="315" src="https://www.youtube.com/embed/4n0xNbfJLR8" frameborder="0" allowfullscreen></iframe>
```

> Sobald Sie die Datei gespeichert haben, schauen Sie auf `/my-files/` Wieder - Die neue Markdown befindet sich in der Tabelle. Dies ist eine sehr leistungsstarke Funktion von Gatsby. Wie im vorherigen Beispiel `siteMetadata` können Quellen-Plugins Daten live neu laden.

Reviewer: ```suggestion
Wenn du die Datei gespeichert hast, öffne erneut `/my-files/` - die neu erstellte Markdown-Datei befindet sich in der Tabelle. Dies ist eine sehr leistungsstarke Funktion von Gatsby. Wie im vorherigen Beispiel `siteMetadata` können Quellen-Plugins Daten live neu laden.
```

> Sobald Sie die Datei gespeichert haben, schauen Sie auf `/my-files/` Wieder - Die neue Abschriftendatei befindet sich in Der Tisch. Dies ist eine sehr leistungsstarke Funktion von Gatsby. Wie im vorherigen Beispiel `siteMetadata` können Quell-Plugins Daten live neu laden.

Reviewer: ```suggestion
Sobald Sie die Datei gespeichert haben, schauen Sie auf `/my-files/` Wieder - Die neue Markdown befindet sich in der Tabelle. Dies ist eine sehr leistungsstarke Funktion von Gatsby. Wie im vorherigen Beispiel `siteMetadata` können Quellen-Plugins Daten live neu laden.
```

> `gatsby-source-filesystem` sucht immer nach neuen Dateien und wann Sie führen Ihre Abfragen erneut aus.

Reviewer: ```suggestion
`gatsby-source-filesystem` sucht kontinuierlich nach neuen Dateien und führt Ihre Abfragen erneut aus, wenn Dateien neu erstellt oder verändert wurden.
```

> Fügen Sie ein Transformer-Plugin hinzu, das Markdown-Dateien umwandeln kann:

Reviewer: ```suggestion
Füge nun ein Transformer-Plugin hinzu, das Markdown-Dateien umwandeln kann:
```

> Fügen Sie ein Transformer-Plugin hinzu, das Markdown-Dateien transformieren kann:

Reviewer: ```suggestion
Fügen Sie ein Transformer-Plugin hinzu, das Markdown-Dateien umwandeln kann:
```

```shell
npm install --save gatsby-transformer-remark
```

> Fügen Sie es dann wie gewohnt in die `gatsby-config.js` ein:

Reviewer: ```suggestion
Fügen Sie es dann, wie gewohnt, in die `gatsby-config.js` Folgendes ein:
```

> Fügen Sie es dann, wie gewohnt, in die `gatsby-config.js` Folgendes ein:

Reviewer: ```suggestion
Füge dann, wie gewohnt, in die `gatsby-config.js` Folgendes ein:
```

}
```

> Starten Sie den Entwicklungsserver neu, aktualisieren Sie GraphiQL (oder öffnen Sie es erneut), und sehen Sie nach bei der Autovervollständigung:

Reviewer: ```suggestion
Starten Sie den Entwicklungsserver neu, aktualisieren Sie GraphiQL (oder öffnen Sie es erneut), und schauen Sie sich die Autovervollständigungen an:
```

> Starten Sie den Entwicklungsserver neu, aktualisieren Sie GraphiQL (oder öffnen Sie es erneut), und schauen Sie sich die Autovervollständigungen an:

Reviewer: ```suggestion
Starte den Entwicklungsserver neu, aktualisiere GraphiQL (oder öffne es erneut), und schau dir dann die Autovervollständigungen an:
```

![markdown-autocomplete](markdown-autocomplete.png)

> Wählen Sie `allMarkdownRemark` erneut aus und führen Sie es wie für `allFile` aus. Du wirst  Dort sehen Sie die Markdown-Datei, die Sie kürzlich hinzugefügt haben. Erforschen Sie die Felder, die sind verfügbar auf dem `MarkdownRemark` Knoten.

Reviewer: ```suggestion
Wählen Sie `allMarkdownRemark` erneut aus und führen Sie es wie für `allFile` aus. Sie wirden dort die Markdown-Datei sehen, die Sie gerade hinzugefügt haben. Schauen Sie sich die Felder, die auf dem `MarkdownRemark` Knoten verfügbar sind, an.
```

> Wählen Sie `allMarkdownRemark` erneut aus und führen Sie es wie für `allFile` aus. Sie wirden dort die Markdown-Datei sehen, die Sie gerade hinzugefügt haben. Schauen Sie sich die Felder, die auf dem `MarkdownRemark` Knoten verfügbar sind, an.

Reviewer: ```suggestion
Wähle `allMarkdownRemark` erneut aus und führe es erneut, wie für `allFile`, aus. Du wirst dort die Markdown-Datei sehen, die du gerade hinzugefügt hast. Schau dir die Felder, die auf dem `MarkdownRemark` Knoten verfügbar sind, an.
```

![markdown-query](markdown-query.png)

> In Ordnung! Hoffentlich fangen einige Grundlagen an, sich zu etablieren. Quell-Plugins bringen Daten _in_ das Gatsby-Datensystem. Dieses Muster kann alle Datenquellen und Datenumwandlungen verarbeiten, die Sie beim Erstellen einer Gatsby-Site benötigen.

Reviewer: ```suggestion
In Ordnung! Hoffentlich fangen einige Grundlagen an, sich zu etablieren. Quellen-Plugins bringen Daten _in_ das Gatsby-Datensystem. Dieses Muster kann alle Datenquellen und -umwandlungen verarbeiten, die Sie beim Erstellen einer Gatsby-Site benötigen.
```

> In Ordnung! Hoffentlich fangen einige Grundlagen an, sich zu etablieren. Quellen-Plugins bringen Daten _in_ das Gatsby-Datensystem. Dieses Muster kann alle Datenquellen und -umwandlungen verarbeiten, die Sie beim Erstellen einer Gatsby-Site benötigen.

Reviewer: ```suggestion
Super! Hoffentlich etablieren sich damit die ersten Grundlagen. Quellen-Plugins bringen Daten _in_ das Gatsby-Datensystem. Mit diesem Schema können alle Datenquellen und -umwandlungen verarbeitet werden, die beim Erstellen einer Gatsby Website benötigt werden.
```

> ## Erstellen Sie eine Liste der Markdown-Dateien Ihrer Site in `src/pages/index.js`

Reviewer: ```suggestion
## Erstell eine Liste der Markdown-Dateien deiner Website in `src/pages/index.js`
```

> Jetzt müssen Sie eine Liste Ihrer Markdown-Dateien auf der Startseite erstellen. Wie viele Blogs, möchten Sie mit einer Liste von Links auf der Titelseite enden, die auf jeden Blogeintrag verlinken. Mit GraphQL können Sie _query_ für die aktuelle Liste der Blog-Beiträge abfragen, sodass Sie die Liste nicht manuell pflegen müssen.

Reviewer: ```suggestion
Jetzt sollst du eine Liste deiner Markdown-Dateien auf der Startseite erstellen. Wie in vielen Blogs, soll auf der Startseite eine Liste von Links, die auf jeden Blogeintrag verlinken, angezeigt werden. Mit GraphQL kannst du _query_ für eine aktuelle Liste von Blog-Beiträge abfragen, sodass die Liste nicht manuell gepflegt werden muss.
```

> Jetzt müssen Sie eine Liste Ihrer Markdown-Dateien auf der Startseite erstellen. Wie viele 87 Blogs, möchten Sie mit einer Liste von Links auf der Titelseite enden, die auf jeden zeigen 88 Blogeintrag. Mit GraphQL können Sie _query_ für die aktuelle Liste der Abschriften Blog 87 Beiträge, sodass Sie die Liste nicht manuell pflegen müssen.

Reviewer: ```suggestion
Jetzt müssen Sie eine Liste Ihrer Markdown-Dateien auf der Startseite erstellen. Wie viele Blogs, möchten Sie mit einer Liste von Links auf der Titelseite enden, die auf jeden Blogeintrag verlinken. Mit GraphQL können Sie _query_ für die aktuelle Liste der Blog-Beiträge abfragen, sodass Sie die Liste nicht manuell pflegen müssen.
```

> Wie bei der `src/pages/my-files.js` Seite ersetzen `src/pages/index.js` mit Im Folgenden wird eine GraphQL-Abfrage mit anfänglichem HTML und Stil hinzugefügt.

Reviewer: ```suggestion
Wie schon bei der Seite `src/pages/my-files.js` ersetze nun in `src/pages/index.js` mit dem Folgenden, um eine eine GraphQL-Abfrage, mit HTML und CSS, hinzuzufügen.
```

`
```

Nun sollte die Startseite so aussehen:

![frontpage](frontpage.png)

> Aber dein einziger Blog-Beitrag sieht ein bisschen einsam aus. Also lasst uns noch einen hinzufügen bei

Reviewer: In case you want to stay with "Sie": then this sentence should become

```suggestion
Aber Ihr einziger Blog-Beitrag sieht ein bisschen einsam aus. Also lassen Sie uns noch einen hinzufügen bei
```

> Aber Ihr einziger Blog-Beitrag sieht ein bisschen einsam aus. Also lassen Sie uns noch einen hinzufügen bei

Reviewer: ```suggestion
Dein Blog-Beitrag sieht aber ein bisschen einsam aus, also lass uns noch einen hinzufügen in
```

![two-posts](two-posts.png)

> Welches sieht gut aus! Außer ... die Reihenfolge der Beiträge ist falsch.

Reviewer: ```suggestion
Das Ergebnis sieht gut aus! Außer dass die Reihenfolge der Beiträge falsch ist.
```

> Das Ergebnis sieht gut aus! Außer dass die Reihenfolge der Beiträge falsch ist.

Reviewer: ```suggestion
Das Ergebnis sieht gut aus! Außer, dass die Reihenfolge der Beiträge nicht stimmt.
```

> Dies ist jedoch leicht zu beheben. Wenn Sie eine Verbindung eines bestimmten Typs abfragen, können Sie der GraphQL-Abfrage eine Reihe von Argumenten übergeben. Sie können Knoten `sort` und `filter`, die Anzahl der zu überspringenden Knoten festlegen und das `limit` für die Anzahl der abzurufenden Knoten festlegen. Mit diesen leistungsstarken Operatoren können Sie beliebige Daten auswählen - in dem Format, das Sie benötigen.

Reviewer: ```suggestion
Dies ist jedoch leicht zu beheben. Wenn Sie eine Verbindung eines bestimmten Typs abfragen, können Sie der GraphQL-Abfrage eine Reihe von Argumenten übergeben. Sie können die Knoten `sort` und `filter`, die Anzahl der zu überspringenden Knoten und das `limit` für die Anzahl der abzurufenden Knoten festlegen. Mit diesen leistungsstarken Operatoren können Sie beliebige Daten auswählen - in dem Format, das Sie benötigen.
```

> Dies ist jedoch leicht zu beheben. Wenn Sie eine Verbindung eines bestimmten Typs abfragen, können Sie der GraphQL-Abfrage eine Reihe von Argumenten übergeben. Sie können die Knoten `sort` und `filter`, die Anzahl der zu überspringenden Knoten und das `limit` für die Anzahl der abzurufenden Knoten festlegen. Mit diesen leistungsstarken Operatoren können Sie beliebige Daten auswählen - in dem Format, das Sie benötigen.

Reviewer: ```suggestion
Dies lässt jedoch leicht beheben. Um eine Verbindung eines bestimmten Typs abzufragen, kannst du der GraphQL-Abfrage eine Reihe von Argumenten übergeben. Du kannst den Knoten mit `sort` und `filter`, die Anzahl der zu überspringenden Knoten und das `limit` für die Anzahl der abzurufenden Knoten festlegen. Mit diesen leistungsstarken Operatoren kannst du beliebige Daten abfragen - in dem Format, das du benötigst.
```

> Ändern Sie in der GraphQL-Abfrage Ihrer Indexseite `allMarkdownRemark` in `allMarkdownRemark` (sort: {fields: [frontmatter___date], order: DESC}). _Hinweis: Zwischen `frontmatter` und` date` gibt es 3 Unterstriche._ Speichern Sie dies und die Sortierreihenfolge sollte festgelegt sein.

Reviewer: ```suggestion
Ändern Sie in der GraphQL-Abfrage Ihrer Indexseite `allMarkdownRemark` in `allMarkdownRemark` (sort: {fields: [frontmatter___date], order: DESC}). _Hinweis: Zwischen `frontmatter` und` date` gibt es 3 Unterstriche._ Speichern Sie die Abfrage und die Sortierreihenfolge sollte nun richtig sein.
```

> Ändern Sie in der GraphQL-Abfrage Ihrer Indexseite `allMarkdownRemark` in `allMarkdownRemark` (sort: {fields: [frontmatter___date], order: DESC}). _Hinweis: Zwischen `frontmatter` und` date` gibt es 3 Unterstriche._ Speichern Sie die Abfrage und die Sortierreihenfolge sollte nun richtig sein.

Reviewer: ```suggestion
Änder in der GraphQL-Abfrage deiner Startseite `allMarkdownRemark` in `allMarkdownRemark` (sort: {fields: [frontmatter___date], order: DESC}). _Hinweis: Zwischen `frontmatter` und` date` sind 3 Unterstriche._ Speiche die Abfrage und nun sollte die Sortierreihenfolge richtig sein.
```

> Versuchen Sie, GraphiQL zu öffnen und mit verschiedenen Sortieroptionen zu spielen. `allFile` lässt sich mit mit anderen Verbindungen sortieren.

Reviewer: ```suggestion
Öffne GraphiQL noch einmal und spiel mit den verschiedenen Sortieroptionen. `allFile` lässt sich mit anderen Verbindungen sortieren.
```

> Versuchen Sie, GraphiQL zu öffnen und mit verschiedenen Sortieroptionen zu spielen. Sie können die sortieren `allFile` Verbindung zusammen mit anderen Verbindungen.

Reviewer: ```suggestion
Versuchen Sie, GraphiQL zu öffnen und mit verschiedenen Sortieroptionen zu spielen. `allFile` lässt sich mit mit anderen Verbindungen sortieren.
```

I don't have an idea how to properly express this in german. I'm also not sure whether `connection` is the right wording here in english? Isn't `allFile` just an attribute for the enduser in that case?

> Weitere Informationen zu unseren Abfrageoperatoren finden Sie in unserer [GraphQL reference guide.](/docs/graphql-reference/)

Reviewer: ```suggestion
Weitere Informationen zu unseren Abfrageoperatoren findest du in unserer [GraphQL Referenz](/docs/graphql-reference/).
```

## Herausforderung

> Versuchen Sie, eine neue Seite mit einem Blogeintrag zu erstellen, und sehen Sie, was mit der Liste der Blogeinträge auf der Startseite geschieht!

Reviewer: ```suggestion
Versuche, eine neue Seite mit einem Blogeintrag zu erstellen, und sieh, was mit der Liste der Blogbeiträge auf der Startseite passiert!
```

## Was kommt als nächstes

> Das ist toll! Sie haben gerade eine schöne Indexseite erstellt, auf der Sie Ihre abgefragten Dateien als eine Liste von Blog-Beitrags-Titeln und Text-Auszügen darstellen. Sie möchten jedoch nicht nur Text-Auszüge sehen, sondern auch tatsächliche Seiten für Ihre Markdown-Dateien.

Reviewer: ```suggestion
Das ist toll! Du hast in diesem Tutoial eine schöne Startseite erstellt, auf der deine abgefragten Dateien als eine Liste von Blogbeitrags-Titeln und Text-Auszügen dargestellt werden. Du willst jedoch nicht nur die Text-Auszüge sehen, sondern auch die tatsächlichen Seiten für deine Markdown-Dateien.
```

> Das ist toll! Sie haben gerade eine schöne Indexseite erstellt, auf der Sie Ihre Abschrift abfragen Dateien und Erstellen einer Liste von Blog-Post-Titeln und Auszügen. Sie möchten jedoch nicht nur Auszüge sehen, sondern auch tatsächliche Seiten für Ihre Markdown-Dateien.

Reviewer: ```suggestion
Das ist toll! Sie haben gerade eine schöne Indexseite erstellt, auf der Sie Ihre abgefragten Dateien als eine Liste von Blog-Beitrags-Titeln und Text-Auszügen darstellen. Sie möchten jedoch nicht nur Text-Auszüge sehen, sondern auch tatsächliche Seiten für Ihre Markdown-Dateien.
```

> Sie können weitere Seiten erstellen, indem Sie die React-Komponenten in `src/pages` platzieren. 

Reviewer: ```suggestion
Du kannst weitere Seiten erstellen, indem du React-Komponenten in `src/pages` erstellst. 
```

> Als Nächstes lernen Sie, wie Sie _programmatisch_ Seiten aus _Daten_ erstellen. 

Reviewer: ```suggestion
Als Nächstes lernst du, wie sich _programmatisch_ Seiten aus _Daten_ erstellen lassen. 
```

> Als Nächstes lernen Sie jedoch, wie Sie _programmatisch_ Seiten aus _Daten_ erstellen. 

Reviewer: ```suggestion
Als Nächstes lernen Sie, wie Sie _programmatisch_ Seiten aus _Daten_ erstellen. 
```

> Gatsby ist _nicht_ darauf beschränkt, Seiten aus Dateien zu erstellen, wie dies bei vielen statischen Site-Generatoren der Fall ist. 

Reviewer: ```suggestion
Gatsby ist allerdings _nicht_ darauf beschränkt, Seiten aus Dateien zu erstellen, wie dies bei vielen statischen Site-Generatoren der Fall ist. 
```

> Mit Gatsby können Sie GraphQL verwenden, um Ihre _Daten_ abzufragen und die Abfrageergebnisse _Seiten_ zuzuordnen - alles zur Erstellungszeit.

Reviewer: ```suggestion
Mit Gatsby kannst du GraphQL verwenden, um deine _Daten_ abzufragen und die Ergebnisse _Seiten_ zuzuordnen - alles zur Erstellungszeit.
```

> Dies ist eine wirklich starke Idee. 

Reviewer: ```suggestion
Dies ist eine wirklich mächtige Idee. 
```

> Im nächsten Lernprogramm werden die Auswirkungen und Verwendungsmöglichkeiten erläutert [programmgesteuert Seiten aus Daten erstellen](/tutorial/part-seven/).

Reviewer: ```suggestion
Im nächsten Tutorial werden die Vorraussetzungen und Möglichkeiten erläutert, mit denen [Seiten programmatisch aus Daten erstellt werden](/tutorial/part-seven/) können.
```
